src/main/java/AutoResponder.java

```diff
@@ -0,0 +1,238 @@
+    static Pattern PATTERN_DEADLINE = Pattern.compile("^deadline (.+) /by (.+)");
+    static Pattern PATTERN_EVENT = Pattern.compile("^event (.+) /at (.+)");
+    static Pattern PATTERN_TODO = Pattern.compile("^todo (.+)");
+    static Pattern PATTERN_DONE = Pattern.compile("^done (\\d+)");
+    static Pattern PATTERN_DELETE = Pattern.compile("^delete (\\d+)");
+    static Pattern PATTERN_EMPTY_COMMAND = Pattern.compile("^(todo|event|deadline|find|done|delete)\\s*$");
+    static Pattern PATTERN_LIST = Pattern.compile("^list\\s*$");
+    static Pattern PATTERN_SAVE = Pattern.compile("^save\\s*$");
+    static Pattern PATTERN_FIND = Pattern.compile("^find (.+)");
+    static Pattern PATTERN_BYE = Pattern.compile("^bye\\s*$");
+
+    private ScrollPane scrollPane;
+    private VBox dialogContainer;
+    private TextField userInput;
+    private Button sendButton;
+    private Scene scene;
+    private Image user = new Image(this.getClass().getResourceAsStream("/images/DaUser.png"));
+    private Image duke = new Image(this.getClass().getResourceAsStream("/images/DaDuke.png"));
+
+
+    /**
+     * Initialiser for empty AutoResponder object.
+     */
+    public AutoResponder() {
+        this.taskList = new ArrayList<>();
+        this.toPrint = new StringBuilder();
+        this.ui = new Ui();
+    }
+
+    private AutoResponder(List<Task> taskList, StringBuilder toPrint) {
+        this.taskList = taskList;
+        this.toPrint = toPrint;
+        this.ui = new Ui();
+    }
+
+    private String printToConsole() {
+        String output = toPrint.toString();
+        this.toPrint = new StringBuilder();
+        return output;
+    }
+
+    /**
+     * Reads input from UI, and processes commands accordingly.
+     *
+     * @param input String to be processed from System.in
+     * @return new AutoResponder object with corresponding command to be processed
+     */
+    public String readInput(String input) {
+        String lowerInput = input.toLowerCase();
+        if (PATTERN_LIST.matcher(lowerInput).find()) {
+            return this.processList();
+        } else if (PATTERN_DONE.matcher(lowerInput).find()) {
+            Matcher m = PATTERN_DONE.matcher(lowerInput);
+            m.find();
+            int index = Integer.parseInt(m.group(1)) - 1;
+            return this.markTaskDone(index);
+        } else if (PATTERN_DELETE.matcher(lowerInput).find()) {
+            Matcher m = PATTERN_DELETE.matcher(lowerInput);
+            m.find();
+            int index = Integer.parseInt(m.group(1)) - 1;
+            return this.deleteTask(index);
+        } else if (PATTERN_SAVE.matcher(lowerInput).find()) {
+            return this.saveList();
+        } else if (PATTERN_EMPTY_COMMAND.matcher(lowerInput).find()) {
+            Matcher m = PATTERN_EMPTY_COMMAND.matcher(lowerInput);
+            m.find();
+            throw new IllegalArgumentException("OOPS!!! The description of a "
+                    + m.group(1) + " cannot be empty.");
```

Comment:
Consider adding exception signatures to the method, it will make the chain of escalating clearer.